tests/nosetests/pyanaconda_tests/dbus_specification_test.py

```diff
@@ -0,0 +1,432 @@
+#
+# Copyright (C) 2019  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Vendula Poncova <vponcova@redhat.com>
+#
+
+import unittest
+
+from pyanaconda.dbus.specification import DBusSpecificationParser, DBusSpecification, \
+    DBusSpecificationError
+
+
+class SpecificationTestCase(unittest.TestCase):
+
+    def from_members_test(self):
+        """Test a specification created from members."""
+        method = DBusSpecification.Method(
+            name="Method",
+            interface_name="A",
+            in_type="()",
+            out_type="(s)"
+        )
+
+        signal = DBusSpecification.Signal(
+            name="Signal",
+            interface_name="B",
+            type="(ii)"
+        )
+
+        prop = DBusSpecification.Property(
+            name="Property",
+            interface_name="B",
+            readable=True,
+            writable=False,
+            type="i"
+        )
+
+        specification = DBusSpecification()
+        specification.add_member(method)
+        specification.add_member(signal)
+        specification.add_member(prop)
+
+        with self.assertRaises(DBusSpecificationError):
+            specification.get_member("A", "Invalid")
+
+        with self.assertRaises(DBusSpecificationError):
+            specification.get_member("Invalid", "Method")
+
+        self.assertEqual(specification.interfaces, ["A", "B"])
+        self.assertEqual(specification.members, [method, signal, prop])
```

Comment:
Can't this fail on order of the values some time?

Reply:
This would regard also 'interfaces' and 'members' properties checks below.

Reply:
The implementation is based on a dictionary and a dictionary keeps in insertion order, so it should be fine.

Reply:
Hmm, since 3.7, I missed that